docs/cluster-agent/HORIZONTAL_POD_AUTOSCALING.md

```diff
@@ -0,0 +1,262 @@
+
+Active connections: 1 
+server accepts handled requests
+ 1 1 1 
+Reading: 0 Writing: 1 Waiting: 0 
+```
+
+Behind the scene, the number of request per second also increased. 
+This metric is being collected by the node agent, as it autodiscovered the nginx pod through its annotations (for more information on how autodiscovery works, see our doc [here](https://docs.datadoghq.com/agent/autodiscovery/#template-source-kubernetes-pod-annotations)).
+Therefore, if you stress it, you will see the uptick in your Datadog app. 
+As you referenced this metric in your HPA manifest, the DCA is also pulling its latest value every 20 seconds.
+Then, as Kubernetes queries the DCA to get this value, it will notice that the number is going above the threshold and will autoscale accordingly.
+
+Let's do it!
+
+Just run `while true; do curl <nginx_svc>:8090/nginx_status; sleep 0.1; done`
+And you should soon see the number of requests per second spiking, going above 9 which is the threshold over which we want to autoscale out nginx boxes.
+Then, you should see new nginx pods being created:
+
+```
+PODS:
+
+NAMESPACE     NAME                                     READY     STATUS    RESTARTS   AGE
+default       datadog-cluster-agent-7b7f6d5547-cmdtc   1/1       Running   0          9m
+default       nginx-6757dd8769-5xzp2                   1/1       Running   0          2m
+default       nginx-6757dd8769-k6h6x                   1/1       Running   0          2m
+default       nginx-6757dd8769-vzd5b                   1/1       Running   0          29m
+
+HPAS:
+
+NAMESPACE   NAME       REFERENCE          TARGETS       MINPODS   MAXPODS   REPLICAS   AGE
+default     nginxext   Deployment/nginx   30/9 (avg)   1         3         3          29m
+
+```
+
+Voila
+
+### Troubleshooting
+
+- Make sure you have the Aggregation layer and the certificates set up as per the requirements section.
+- Always make sure the metrics you want to autoscale on are available
+As you create the HPA, the DCA will parse the manifest and query Datadog to try to fetch the metric.
+If there is a typo or if the metric does not exist you will see an error such as:
+```
+2018-07-03 13:47:56 UTC | ERROR | (datadogexternal.go:45 in queryDatadogExternal) | Returned series slice empty
+```
+You can check in the Configmap used to store and share the HPA state by the DCA:
+`kubectl get cm datadog-hpa -o yaml`
+will yield:
+```
+apiVersion: v1
+data:
+  external.metrics.default.nginxext-nginx.net.request_per_s: '{"name":"nginx.net.request_per_s","labels":{"kube_container_name":"nginx"},"ts":1530625676,"hpa_name":"nginxext","hpa_namespace":"default","value":0,"valid":false}'
+kind: ConfigMap
+metadata:
+  creationTimestamp: 2018-07-03T13:40:50Z
+  name: datadog-hpa
+  namespace: default
+  resourceVersion: "1742"
+```
+Here you can see if the metric has a typo or check that it exists in Datadog. If the metric's flag `Valid` is set to false, the metric will not be considered in the HPA pipeline.
```

Comment:
r/`has a typo or check that it exists in Datadog`/`has a typographic issue or check if it exists in Datadog`

Reply:
r/`If the metric's flag `Valid` is set to false, the metric will not be considered in the HPA pipeline.
`/`If the metric's flag `Valid` is set to false, the metric is not considered in the HPA pipeline.`